src/components/PageDefinitions.js

```diff
@@ -46,29 +46,55 @@ class PageDefinitions extends Component {
     dispatch(uiBrowseUpdateList({ add: component }))
   }
 
-  onDrop(acceptedFile, rejectedFiles) {
+  onDrop(acceptedFiles, rejectedFiles) {
     const { dispatch, token, definitions } = this.props
-    acceptedFile.forEach(file => {
-      const reader = new FileReader();
+    dispatch(uiNotificationNew({ type: 'info', message: 'Loading component list from file(s)', timeout: 5000 }))
+    acceptedFiles.forEach(file => {
+      const reader = new FileReader()
       reader.onload = () => {
-        const components = JSON.parse(reader.result);
-        if (components.coordinates) {
-          components.coordinates.forEach(component => {
-            const spec = EntitySpec.validateAndCreate(component)
-            if (spec) {
-              const path = spec.toPath()
-              !definitions.entries[path] && dispatch(getDefinitionsAction(token, [path]))
-              dispatch(uiBrowseUpdateList({ add: spec }))
-            }
-          })
-        } else {
-          dispatch(uiNotificationNew({ type: 'info', message: 'Invalid file.', timeout: 5000 }))
+        const listSpec = this.loadListSpec(reader.result, file)
+        if (typeof listSpec === 'string') {
+          const message = `Invalid component list file: ${listSpec}`
+          return dispatch(uiNotificationNew({ type: 'info', message, timeout: 5000 }))
         }
-      };
+        listSpec.coordinates.forEach(component => {
+          // TODO figure a way to add these in bulk. One by one will be painful for large lists
+          const spec = EntitySpec.validateAndCreate(component)
+          if (spec) {
+            const path = spec.toPath()
+            !definitions.entries[path] && dispatch(getDefinitionsAction(token, [path]))
+            dispatch(uiBrowseUpdateList({ add: spec }))
+          }
+        })
+      }
       reader.readAsBinaryString(file)
     })
   }
 
+  loadListSpec(content, file) {
+    try {
+      const object = JSON.parse(content)
+      if (file.name.toLowerCase() === 'package-lock.json') return this.loadPackageLockFile(object.dependencies)
+      if (object.coordinates) return object
+      return 'No component coordinates found'
+    } catch (e) {
+      return e.message
+    }
+  }
+
+  loadPackageLockFile(dependencies) {
+    const coordinates = []
+    for (const dependency in dependencies) {
```

Comment:
Change the for loop to a forEach to make more readable and match style of remaining project